dyncall/dyncall_registry.go

```diff
@@ -0,0 +1,86 @@
+package dyncall
+
+import "fmt"
+
+type getCallModule struct {
+	respChan chan CallModule
+	id       string
+}
+
+type Registry struct {
+	addModuleChan chan CallModule
+	getModuleChan chan getCallModule
+	closeChan     chan struct{}
+}
+
+func New() *Registry {
+
+	r := &Registry{
+		addModuleChan: make(chan CallModule),
+		getModuleChan: make(chan getCallModule),
+		closeChan:     make(chan struct{}),
+	}
+
+	// state
+	go func() {
+		modules := map[string]CallModule{}
+
+		for {
+			select {
+			case <-r.closeChan:
+				return
+			case m := <-r.addModuleChan:
+				modules[m.CallID()] = m
+			case getCallMod := <-r.getModuleChan:
+				mod := modules[getCallMod.id]
```

Comment:
We should probably check if map key exists first to avoid any potential panics

The current implementation might still work correctly in practice though, even without any changes

In the case of the key not existing in the modules map, `nil` will be sent to `getCallMod.respChan` so as long as it's properly handled, it shouldn't cause major issues

A proposed alternative that feels more safe in terms of potential panics
```
if mod, exists := modules[getCallMod.id]; exists {
    getCallMod.respChan <- mod
} else {
    getCallMod.respChan <- nil
}

```
Even though it might not make much of a difference in this use case, if `CallModule` was a struct with 2 fields, of type int and string, you wouldn't get a `nil` if the key is missing in the map with the unsafe usage, and you would get {0,"") as this would be corresponding to the default values of int and string type. 

It's just that this time the default value of `CallModule` is `nil` because its an interface